src/svmlin.cpp

```diff
@@ -0,0 +1,129 @@
+/*    Copyright 2006 Vikas Sindhwani (vikass@cs.uchicago.edu)
+ SVM-lin: Fast SVM Solvers for Supervised and Semi-supervised Learning
+
+ This file is part of SVM-lin.
+
+ SVM-lin is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2 of the License, or
+ (at your option) any later version.
+
+ SVM-lin is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with SVM-lin (see gpl.txt); if not, write to the Free Software
+ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#include <Rcpp.h>
+#include <iostream>
+#include <stdlib.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <cstring>
+#include "ssl.h"
+using namespace Rcpp;
+
+struct options *Options = new options[1];
+struct data *Data = new data[1];
+struct vector_double *Weights = new vector_double[1];
+struct vector_double *Outputs = new vector_double[1];
+struct vector_double *Labels = new vector_double[1];
+
+// [[Rcpp::export]]
+List svmlin_rcpp(S4 X,
+                 NumericVector y,
+                 int l,
+                 int algorithm,
+                 double lambda,
+                 double lambda_u,
+                 int max_switch,
+                 double pos_frac,
+                 double Cp,
+                 double Cn,
+                 NumericVector costs,
+                 bool verbose) {
+  // Set options
+  Options->algo = algorithm;
+  Options->lambda=lambda;
+  Options->lambda_u=lambda_u;
+  Options->S=max_switch;
+  Options->R=pos_frac;
+  Options->epsilon=EPSILON;
```

Comment:
We should have a common definition of epsilon for all our packages in the quanteda header.